itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenarios.java

```diff
@@ -589,6 +590,72 @@ public void testIncrementalAdds() throws IOException {
   }
 
   @Test
+  public void testIncrementalLoadWithVariableLengthEventId() throws IOException, TException {
```

Comment:
well if we cant detect the affect of alter(99) then might be better to write the test more tightly coupled with the number of events that insert generates ( 3 here ) . as number of events per insert might be more than 3 say about 7, with first 5 doing things around stat and as such then we face the same problem as with my previous example, but with the code below. one way to make it more robust might be to, in the behavior injection for event id we explicitly check for the truncate event and give it and id of 100 with all else being an increment from 1.